joblib/test/test_memory.py

```diff
@@ -1045,6 +1045,14 @@ def test_memory_reduce_size_age_limit(tmpdir):
     assert not set.issubset(set(cache_items), set(ref_cache_items))
     assert len(cache_items) == 2
 
+    # ensure age_limit is forced to be positive
+    is_enforced = False
+    try:
+        memory.reduce_size(age_limit=datetime.timedelta(seconds=-1))
+    except AssertionError:
+        is_enforced = True
+    assert is_enforced
```

Comment:
Use the [pytest functionality](https://docs.pytest.org/en/stable/getting-started.html#assert-that-a-certain-exception-is-raised) for better error reporting:
```suggestion
    with pytest.raises(ValueError, match="has to be a positive"):
        memory.reduce_size(age_limit=datetime.timedelta(seconds=-1))
```